kizhin.evgeniy/K2

sdjklfjsdlkfjsdklfj opened the pull request.

Commit:
Implement forward list initialization

Commit:
Introduce insertDuplicates function

Commit:
Introduce output function

Commit:
Implement main

Commit:
Move clear to public interface. Fix leak of list in main

Commit:
Update position computing: begin with 1

Commit:
Merge Volgarenok/spbspu-labs-2025-aads-classes-b master

Volgarenok commented on kizhin.evgeniy/K2/forward-list.hpp:
```diff
@@ -0,0 +1,20 @@
+    FwdList* next;
+  };
+
+  FwdList* initList(int, int);
+  FwdList* insertDuplicates(FwdList*, std::size_t, std::size_t);
+  void clear(FwdList*) noexcept;
+  std::ostream& outputList(std::ostream&, FwdList*);
```
Ну всё таки если у вас два параметра одинакового типа, то неплохо было бы указать их имена

Volgarenok commented on kizhin.evgeniy/K2/forward-list.cpp:
```diff
@@ -0,0 +1,76 @@
+
+void kizhin::clear(FwdList* head) noexcept
+{
+  if (head == nullptr) {
+    return;
+  }
```
Если чуть удачней написать эту функцию, то отдельно эта проверка будет не нужна

Commit:
Refactor insertDuplicates and clear functions

- Removed extra condition in clear
- Simplified insertDuplicates functions by delegating node search to
  getNodeByIndex function
- Fixed issue with uncaught exception in main

Commit:
Refactor main: separately getNodeByIndex call

Commit:
Add const in output function. Refactor getNodeByIndex

Commit:
Merge master of Volgarenok/spbspu-labs-2025-aads-classes-b

Commit:
Merge master of Volgarenok/spbspu-labs-2025-aads-classes-b
